client/basic-authorship/src/basic_authorship.rs

```diff
@@ -38,21 +39,41 @@ use futures::{executor, future, future::Either};
 use sp_blockchain::{HeaderBackend, ApplyExtrinsicFailed::Validity, Error::ApplyExtrinsicFailed};
 use std::marker::PhantomData;
 
+use prometheus_endpoint::Registry as PrometheusRegistry;
+use sc_proposer_metrics::MetricsLink as PrometheusMetrics;
+
 /// Proposer factory.
 pub struct ProposerFactory<A, B, C> {
 	/// The client instance.
 	client: Arc<C>,
 	/// The transaction pool.
 	transaction_pool: Arc<A>,
+	/// Prometheus Link,
+	metrics: PrometheusMetrics,
```

Comment:
Out of curiosity what is the concept of a `Link`? Like a reference? Do we refer to the Prometheus metrics as `Prometheus Link` anywhere else?

Reply:
this is not directly to the prometheus, it is a link to metrics
which is *declared* as MetricsLink